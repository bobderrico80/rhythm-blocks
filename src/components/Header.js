import React from 'react';
import PropTypes from 'prop-types';
import styles from './Header.module.css';

const propTypes = {
  onAboutButtonClick: PropTypes.func.isRequired,
};

const Header = ({ onAboutButtonClick }) => (
  <header className={styles.header}>
    <h1 className={styles.banner}>Rhythm Blocks</h1>
    <nav className={styles.nav}>
      <ul className={styles.navList}>
        <li className={styles.navItem}>
          <button className={styles.navButton} onClick={onAboutButtonClick}>
            About
          </button>
        </li>
      </ul>
    </nav>
  </header>
);

Header.propTypes = propTypes;

export default Header;
