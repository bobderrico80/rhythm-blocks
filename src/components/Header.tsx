import React from 'react';
import PropTypes from 'prop-types';
import styles from './Header.module.css';
import icon from '../assets/svg/ui/rb-icon.svg';

export interface HeaderProps {
  onAboutButtonClick: (event: any) => void;
}

const Header = ({ onAboutButtonClick }: HeaderProps) => (
  <header className={styles.header}>
    <h1 className={styles.banner}>
      <img src={icon} alt="" />
      Rhythm Blocks
      <small className={styles.versionText}>Alpha</small>
    </h1>
    <nav className={styles.nav}>
      <ul className={styles.navList}>
        <li className={styles.navItem}>
          <button className={styles.navButton} onClick={onAboutButtonClick}>
            About
          </button>
        </li>
      </ul>
    </nav>
  </header>
);

export default Header;
