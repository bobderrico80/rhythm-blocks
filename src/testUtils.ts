import { ReactElement } from 'react';
import renderer from 'react-test-renderer';

export const createSnapshot = (component: ReactElement<any>) => {
  const snapshot = renderer.create(component);
  return snapshot.toJSON();
};
