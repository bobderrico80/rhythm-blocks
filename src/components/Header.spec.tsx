import React, { ReactElement } from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { createSnapshot } from '../testUtils';
import Header from './Header';

describe('The <Header /> component', () => {
  let component: ReactElement<any>;
  let wrapper: ShallowWrapper;
  let onAboutButtonClick: jest.Mock;

  beforeEach(() => {
    onAboutButtonClick = jest.fn();
    component = <Header onAboutButtonClick={onAboutButtonClick} />;
    wrapper = shallow(component);
  });

  it('renders as expected', () => expect(createSnapshot(component)).toMatchSnapshot());

  it('passes onAboutButtonClick prop to About button onClick attribute', () => {
    expect(wrapper.find('button').prop('onClick')).toEqual(onAboutButtonClick);
  });
});
