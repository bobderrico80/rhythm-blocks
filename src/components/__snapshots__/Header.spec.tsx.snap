// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <Header /> component renders as expected 1`] = `
<header
  className="header"
>
  <h1
    className="banner"
  >
    <img
      alt=""
      src="rb-icon.svg"
    />
    Rhythm Blocks
    <small
      className="versionText"
    >
      Alpha
    </small>
  </h1>
  <nav
    className="nav"
  >
    <ul
      className="navList"
    >
      <li
        className="navItem"
      >
        <button
          className="navButton"
          onClick={[MockFunction]}
        >
          About
        </button>
      </li>
    </ul>
  </nav>
</header>
`;
